src/compiler/checker.ts

```diff
@@ -48894,6 +48909,37 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
     }
 
+    function getNonlocalEffectiveTypeAnnotationNode(node: Node) {
+        const direct = getEffectiveTypeAnnotationNode(node);
+        if (direct) {
+            return direct;
+        }
+        if (node.kind === SyntaxKind.Parameter && node.parent.kind === SyntaxKind.SetAccessor) {
+            const other = getAllAccessorDeclarationsForDeclaration(node.parent as SetAccessorDeclaration).getAccessor;
+            if (other) {
+                return getEffectiveReturnTypeNode(other);
+            }
+        }

```

Comment:
Just to be clear, this code is a noop, right? Since we're not actually handling these and no baselines change? This is brand new code, so, checking.

Reply:
Yes, for now. My goal is to move the full "should this node be reused" logic check into the node builder, to minimize the `EmitResolver`'s API surface. Today, the node builder never has to emit nonlocal types for `get` and `set` accessors. If one is missing an annotation, it gets the type the other one has, and if both have the same type, it prints them as a property, rather than accessors. Meanwhile, declaration emit proper obviously leaves accessor declarations as-is nowadays, and uses logic like this to pull the type node from the other accessor when one is missing an annotation. So this is *slightly* future-looking, in that it enables the node builder to properly reuse nodes for `get`/`set` pairs that are missing one type node, even though the declaration emitter doesn't *yet* ask it to do that (and you can't get an inferred type with a `get`/`set` pair without them having differing types, and thus both annotated).